### PR TITLE
refactor(ast-evaluation): return RootNodeEvaluation instead of bool

### PR DESCRIPTION
**Pull request by balzdur**
First PR to expose `NodeEvaluation` to scenario evaluation.

I prefer to show you this refactor before moving forward.

### CODE REVIEW

File: usecases/evaluate_scenario/evaluate_scenario.go [hunk 1]
```diff
@@ -207,34 +200,56 @@ func evalScenarioRule(ctx context.Context, repositories ScenarioEvaluationReposi
...
 		payload,
 		dataModel,
 	)
+	isAuthorizedError := models.IsAuthorizedError(err)
+	if err != nil && !isAuthorizedError {
+		return errors.Wrap(err,
+			"Unexpected error evaluating trigger condition in EvalScenario")
+	}
+
+	if !triggerEvaluation.ReturnValue || isAuthorizedError {
```
[the remark this answers is not in the document]
Comment (Pascal-Delange) [reply usecases/evaluate_scenario/evaluate_scenario.go:238]: hmm ok, though I still don't totally see how this makes our life a lot better 🤔

**Review by Pascal-Delange — APPROVED** [review]
I didn't totally understand how the new sugar type makes our life easier. That said I'm not particularly opposed to it either (it's easy enough to phase out if needed anyway).

File: usecases/evaluate_scenario/evaluate_scenario.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (balzdur) [reply usecases/evaluate_scenario/evaluate_scenario.go:238]: In short:
Before that, the cast + error handling to get a `bool` return value was done inside `EvaluateAstExpression`. Now we return the whole `NodeEvaluation` we need to cast the return value to a `bool` somewhere, I could have done that :
- inside the `EvaluateAstExpression` and return a `(bool, EvaluateAstExpression, error)` tuple
- directly where it's needed after calling `EvaluateAstExpression` (with inline code or a utility function)
- using a sugar type

[the remark this answers is not in the document]
Comment (Pascal-Delange) [reply usecases/evaluate_scenario/evaluate_scenario.go:238]: yep all those work fine for me 👍

**Pull request merged**